### ISSUE TITLE
train model

### ISSUE THREAD

**Issue by fsted** [gaviotas/AMN#1]
hello ,thanks for your code,Will you make the training code public?
I want to train it

**Comment by gaviotas** [comment]
Hello, @fsted 

You can train AMN by uncommenting lines 18-21 in [generate_pseudo_mask.sh](https://github.com/gaviotas/AMN/blob/7f612d00e434eab7766ec517473cdf1b9d639192/scripts/generate_pseudo_mask.sh#L18).
And the training code is available in [step/train_amn.py](https://github.com/gaviotas/AMN/blob/7f612d00e434eab7766ec517473cdf1b9d639192/step/train_amn.py).

Please check it and let me know if you have any further questions :)

**Comment by fsted** [comment]
> 你好，@fsted
> 
> [您可以通过取消注释generate_pseudo_mask.sh](https://github.com/gaviotas/AMN/blob/7f612d00e434eab7766ec517473cdf1b9d639192/scripts/generate_pseudo_mask.sh#L18)中的第 18-21 行来训练 AMN 。 训练代码在[step/train_amn.py](https://github.com/gaviotas/AMN/blob/7f612d00e434eab7766ec517473cdf1b9d639192/step/train_amn.py)中可用。
> 
> 请检查它，如果您有任何其他问题，请告诉我:)

thank you for your reply! I successfully reproduced the accuracy, but when I try to replace the backbone of amn with resnet101, the accuracy is reduced; what do I need to do to replace the backbone with resnet101?Looking forward to your reply！

**Comment by gaviotas** [comment]
Sorry for the late reply.

Unfortunately, we haven't experimented with AMN with ResNet101.
So, it is difficult to give some advice on replacing the backbone with ResNet101.

Perhaps there is a problem with overfitting the noise of the target supervision (CAM+CRF) as the complexity of the backbone increases. I think that the noise problem should be handled to work with ResNet101.

Thanks.

**Comment by fsted** [comment]
> 这么晚才回复很抱歉。
> 
> 不幸的是，我们还没有用 ResNet101 试验过 AMN。 因此，很难就用 ResNet101 替换主干给出一些建议。
> 
> 随着主干复杂度的增加，目标监督（CAM+CRF）的噪声可能会出现过度拟合的问题。我认为应该处理噪声问题以使用 ResNet101。
> 
> 谢谢。

Thanks for your answer, I think it's worth a try; this is the cleanest parallelized code I've seen, thanks for sharing.

**Comment by fsted** [comment]
> Sorry for the late reply.
> 
> Unfortunately, we haven't experimented with AMN with ResNet101. So, it is difficult to give some advice on replacing the backbone with ResNet101.
> 
> Perhaps there is a problem with overfitting the noise of the target supervision (CAM+CRF) as the complexity of the backbone increases. I think that the noise problem should be handled to work with ResNet101.
> 
> Thanks.

Thanks for your answer, I think it's worth a try; this is the cleanest parallelized code I've seen, thanks for sharing. (∩_∩)
